test/gtest/ucp/test_ucp_am.cc

```diff
@@ -433,6 +433,7 @@ class test_ucp_am_nbx : public test_ucp_am_base {
             param.memh          = memh;
         }
 
+        m_am_sent_counter++;
```

Comment:
Move to after `send`

Reply:
I have fear of concurrencies issues for example AM CB is called before sent_counter is advanced. it can cause the received counter to be larger than sent counter, and I think the invariant should be that received counter is always <= sent counter.
Maybe it's not causing issues now, but it can cause issues in the future, for example if we choose to replace the m_am_received flag we will probably compare the values of sent and receive counters in the CB.

Reply:
better keep it before sending, but rename to m_am_send_counter